tools/pg_tools/macros.xml

```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0"?>
+<macros>
+    <xml name="inputs_outputs">
+        <inputs>
+            <param format="tabular" help="" label="Input tabular dataset" name="in_file" type="data" />
+            <yield />
+        </inputs>
+        <outputs>
+            <data format="tabular" name="out_file" label="${tool.name} on ${on_string}" />
+        </outputs>
+    </xml>
+
+    <xml name="requirements">
+        <requirements>
+            <requirement type="package" version="@PG_VERSION@">postgresql</requirement>
+        </requirements>
+    </xml>
+
+    <token name="@PG_VERSION@">9.6.3</token>
+
+    <token name="@HELP_FOOTER@">
+    The postgresql tools, are using a postgresql database in user-space. Databases are packed into a tarball and stored in the user-history.
+    </token>
+
+    <token name="@UNTAR_INFILE@"><![CDATA[
+        tar -xJvf '$infile' &&
+        export PWD=`pwd` &&
+        sed -i.bak "s|unix_socket_directories.*|unix_socket_directories = '\$PWD/postgresql/'|" ./postgresql/db/postgresql.conf
+    ]]>
+    </token>
+
+    <token name="@PG_SETUP@"><![CDATA[
+            '$__tool_directory__/pglite' setup  -d ./postgresql &&
+            @PG_START@
+    ]]>
+    </token>
+
+    <token name="@PSQL@"><![CDATA[
+        psql "\$('$__tool_directory__/pglite' url -d ./postgresql)"
+    ]]>
+    </token>
+
+
+    <token name="@PG_START@"><![CDATA[
+        '$__tool_directory__/pglite' start  -d ./postgresql &&
+        sleep 10
+    ]]>
+    </token>
+
+    <token name="@PG_STOP@"><![CDATA[
+        '$__tool_directory__/pglite' stop  -d ./postgresql &&
+        sleep 10
+    ]]>
+    </token>
+
+    <token name="@ARCHIVE_DATABASE@"><![CDATA[
+        tar -cvJf postgresql_out.tar.xz ./postgresql
```

Comment:
use the html extra files path thing to avoid the overhead of tarring / un-tarring.

Reply:
Retrated: out of band discussion concluded that disk space was more important than the computational overhead.